### ISSUE TITLE
Write from SQLite to TSV

### ISSUE THREAD

**Issue by jamesaoverton** [jamesaoverton/cmi-pb-terminology#36]
We have code to load from TSV into SQL. Now we need code to write from SQL to TSV. The 'table' table contains information such as the file path to write, and we can add more information as needed. This should be fairly straightforward, but there are little things that I worry about:

1. TSV formatting: I want to output "clean" TSV, like we are working on in [tsvalid](https://github.com/ontodev/tsvalid)
    - Unix line endings (`\n`) to separate rows
    - tabs to separate columns
    - no quotation of cell values
    - inside cells, escape newlines with `\n` and tabs with `\t`
    - same number of columns for each row: if a row ends with empty cells, then in the TSV that line should end with tabs (trailing whitespace)
2. Sorting: We will be storing TSV in version control, and we want version control diffs to include only relevant changes in table content, not "spurious" changes in sorting and formatting. This may require adding more information to the 'table' table about sorting, or making use of primary key information from the 'column' table, or establishing sorting defaults, or all these things.
    - for tables with primary keys, then I think we want to sort by primary key first, then all other columns from left to right
    - for tables without primary keys, I think we want to use SQL row numbering to preserve order

The goal is to have very clean roundtrips from TSV to SQL and back.

**Comment by jamesaoverton** [comment]
There's another part to this that I failed to articulate. I apologize. It's been a confusing week.

It's good to be able to output the _meta columns and their JSON (and we should keep that functionality), but the main use case is "round-tripping": load TSV into SQL, modify the SQL database, then store back to TSV. The resulting TSV should be the same as the original TSV, modulo specific changes to the SQL database. In the base case where the original TSV was already properly formatted and sorted, and there no changes made to the SQL database, the output TSV should be **exactly** the same as the input TSV, even if the cells contain invalid data.

What I failed to articulate is that for this use case the output TSV:

1. should not include _meta columns
2. should include cell values even if they are invalid
3. should include the conflict tables

So I think we need to keep the TSV writing functionality currently in #42, and build on it to merge `value`s from the _meta JSON into the valid value columns.
